eth_defi/aave_v3/events.py

```diff
@@ -0,0 +1,311 @@
+            "csv_writer": csv_writer,
+        }
+
+    log_info(f"Scanning block range {restored_start_block:,} - {end_block:,}")
+    with tqdm(total=end_block - restored_start_block) as progress_bar:
+        #  1. update the progress bar
+        #  2. save any events in the buffer in to a file in one go
+        def update_progress(
+            current_block,
+            start_block,
+            end_block,
+            chunk_size: int,
+            total_events: int,
+            last_timestamp: int,
+            context: TokenCache,
+        ):
+            nonlocal buffers
+
+            if last_timestamp:
+                # Display progress with the date information
+                d = datetime.datetime.utcfromtimestamp(last_timestamp)
+                formatted_time = d.strftime("%d-%m-%Y")
+                progress_bar.set_description(f"Block: {current_block:,}, events: {total_events:,}, time:{formatted_time}")
+            else:
+                progress_bar.set_description(f"Block: {current_block:,}, events: {total_events:,}")
+
+            progress_bar.update(chunk_size)
+
+            # Update event specific contexes
+            for buffer_data in buffers.values():
+                buffer = buffer_data["buffer"]
+
+                # log_info(f'Writing buffer to file {len(buffer)} events')
+                # write events to csv
+                for entry in buffer:
+                    buffer_data["csv_writer"].writerow(entry)
+                    buffer_data["total"] += 1
+
+                # then reset buffer
+                buffer_data["buffer"] = []
+
+            # Sync the state of updated events
+            state.save_state(current_block)
+
+        # Read specified events in block range
+        for log_result in read_events_concurrent(
+            executor,
+            restored_start_block,
+            end_block,
+            events=contract_events,
+            notify=update_progress,
+            chunk_size=100,
+            context=token_cache,
+        ):
+            try:
+                # write to correct buffer
+                event_name = log_result["event"].event_name
+                buffer = buffers[event_name]["buffer"]
+                decode_function = event_mapping[event_name]["decode_function"]
+                decoded_result = decode_function(aave_network_name, log_result)
+                # Note: decoded_result is None if the event is e.g. from Aave v2 contract
+                if decoded_result:
+                    logger.debug(f'Adding event to buffer: {event_name}')
+                    buffer.append(decoded_result)
+            except Exception as e:
+                raise RuntimeError(f"Could not decode {log_result}") from e
+
+    # Write remaining events, close files and print stats
+    for event_name, buffer in buffers.items():
+        if len(buffer["buffer"]) > 0:
+            # log_info(f'Writing buffer to file {len(buffer["buffer"])} events')
```

Comment:
The same, this log call not needed anymore?

Reply:
Probably some commented-out log calls left around here and there after debugging, I can remove them where I see them.